src/main/java/client/Node.java

```diff
@@ -159,7 +160,14 @@ public boolean isInitialState() {
 	}
 
 	public boolean isGoalState() {
```

Comment:
This might change the semantics of the code up to a point where stuff developed for SA might not work properly anymore (unless corresponding changes implemented). Just a note, so I don't forget - I still need to look into it some more.